Packages/com.unity.multiplayer.samples.coop/Utilities/Net/NetworkObjectSpawner.cs

```diff
@@ -0,0 +1,137 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using Unity.Netcode;
+using UnityEditor;
+using UnityEditor.SceneManagement;
+using UnityEngine;
+
+namespace Unity.Multiplayer.Samples.Utilities
+{
+    /// <summary>
+    /// Custom spawning component to be added to a scene GameObject. This component collects NetworkObjects in a scene
+    /// marked by a special tag, collects their Transform data, destroys their prefab instance, and performs the dynamic
+    /// spawning of said objects during Netcode for GameObject's (Netcode) OnNetworkSpawn() callback.
+    /// </summary>
+    [RequireComponent(typeof(NetcodeHooks))]
```

Comment:
Are there chances this object will get its client or server version stripped?
Why the Netcode hooks here?

Reply:
The spawning logic should only operate on the server, and we need NetcodeHooks to know when this in-scene placed NetworkObject is spawned by the network.

Would renaming `ServerNetworkObjectSpawner` help here?

Reply:
what I meant is why not have this be a NetworkBehaviour directly?

Reply:
That is a good question. There shouldn't be any other MonoBehaviours to listen for callbacks. Will refactor.